source/grid/grid_tools.cc

```diff
@@ -2645,16 +2645,43 @@ namespace GridTools
     // Get cell weighting if a signal has been attached to the triangulation
     if (!triangulation.signals.cell_weight.empty())
```

Comment:
What about solving that issue by introducing a new signal `pre_partitioning` that will be triggered in the beginning of any `GridTools::partition_*` function?

Reply:
Sure, signals are always a good idea.

Reply:
Your changes in #8056 make these signals obsolete. I like that :-)